### PR TITLE
Adds ETAS links using the HathiTrust BibAPI

### PR DESCRIPTION
**Pull request by cdmo**
#54

### CODE REVIEW

File: spec/controllers/etas_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe EtasController, type: :controller do
+  describe '#show' do
+    before do
+      json = file_fixture('21383783.json').read
```
Comment (cdmo) [inline spec/controllers/etas_controller_spec.rb:8]: see https://relishapp.com/rspec/rspec-rails/v/3-5/docs/file-fixture

File: spec/features/ask_a_librarian_spec.rb [hunk 2]
```diff
@@ -4,9 +4,13 @@
...
 RSpec.feature 'Ask a librarian', type: :feature do
   describe 'side tab widget', js: true do
+    before do
+      stub_request(:any, /hathitrust/).to_return(status: 200, body: '{}', headers: {})
+    end
+
     it 'shows up on the homepage' do
       visit root_path
-      expect(page).to have_css('button.libchat_online')
+      expect(page).to have_css('button[class^="libchat"]')
```
Comment (banukutlu) [inline spec/features/ask_a_librarian_spec.rb:13]: thanks for fixing this ✔️

File: app/helpers/catalog_helper.rb [hunk 3]
```diff
@@ -98,6 +98,10 @@ def render_thumbnail(document, _options = {})
     end
   end
 
+  def oclc_number(document, _options = {})
```
Comment (banukutlu) [inline app/helpers/catalog_helper.rb:101]:
```suggestion
  def oclc_number(document)
```

File: app/javascript/hathi/index.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const etasSlots = document.querySelectorAll("[data-etas-oclc]");
+
+const hathiETAS = () => {
+    etasSlots.forEach(async (etasSlot) => {
+        const oclcNumberFirst = JSON.parse(etasSlot.dataset.etasOclc);
```
[the remark this answers is not in the document]
Comment (cdmo) [reply app/javascript/hathi/index.js]: this is in

File: app/controllers/etas_controller.rb [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    def hathitrust_etas_url(hathi_hash)
+      hathi_hash.fetch('items').first.fetch('itemURL')
+    end
+
+    def hathitrust_response_hash
+      @hathitrust_response_hash ||= JSON.parse hathitrust_api_response
+    end
+
+    # rubocop:disable MethodLength
+    def hathitrust_api_response
```
[the remark this answers is not in the document]
Comment (cdmo) [reply app/controllers/etas_controller.rb:44]: I agree. When we do more with Hathi (like non ETAS times Hathi) lets circle back to this.

File: spec/controllers/etas_controller_spec.rb (hunk 1 above)
Comment (banukutlu) [reply spec/controllers/etas_controller_spec.rb:8]: 👌

**Pull request merged**